src/test/java/uk/ac/ebi/checklistconverter/ChecklistConverterServiceBiosamplesTest.java

```diff
@@ -0,0 +1,192 @@
+      File file = resourceLoader.getResource("classpath:samples/ERC000011/SAMEA115394517-bsd.json").getFile();
+      JsonNode sampleJson = objectMapper.readValue(file, JsonNode.class);
+
+      requestNode = objectMapper.createObjectNode();
+      requestNode.set("data", sampleJson);
+      requestNode.set("schema", schemaJson);
+
+      response = webClient.post()
+          .uri("http://localhost:3020/validate")
+          .bodyValue(requestNode)
+          .retrieve()
+          .toEntity(JsonNode.class)
+          .toFuture()
+          .get();
+    } catch (IOException | InterruptedException | ExecutionException e) {
+      throw new RuntimeException(e);
+    }
+
+    if (response.getBody() == null) {
+      throw new TestAbortedException("Response body can not be empty");
+    }
+    assertEquals(0, response.getBody().size());
+  }
+
+  @Test
+  void invalid_json_should_output_schema_error() {
+    ResponseEntity<JsonNode> response;
+
+    try {
+      ObjectNode requestNode;
+      String schema = checklistConverterService.getChecklist("ERC000011");
+      JsonNode schemaJson = objectMapper.readValue(schema, JsonNode.class);
+      File file = resourceLoader.getResource("classpath:samples/ERC000011/invalid_geo_location-bsd.json").getFile();
+      JsonNode sampleJson = objectMapper.readValue(file, JsonNode.class);
+
+      requestNode = objectMapper.createObjectNode();
+      requestNode.set("data", sampleJson);
+      requestNode.set("schema", schemaJson);
+
+      response = webClient.post()
+          .uri("http://localhost:3020/validate")
+          .bodyValue(requestNode)
+          .retrieve()
+          .toEntity(JsonNode.class)
+          .toFuture()
+          .get();
+    } catch (IOException | InterruptedException | ExecutionException e) {
+      throw new RuntimeException(e);
+    }
+
+    if (response.getBody() == null) {
+      throw new TestAbortedException("Response body can not be empty");
+    }
+    assertTrue(response.getBody().get(0).get("errors").get(0).asText().contains("must be equal to one of the allowed values"));
+  }
+
+  @Test
+  void synonyms_should_be_converted_correctly_for_field_names() {
+    ResponseEntity<JsonNode> response;
+
+    try {
+      ObjectNode requestNode;
+      String schema = checklistConverterService.getChecklist("ERC000024");
+      JsonNode schemaJson = objectMapper.readValue(schema, JsonNode.class);
+      File file = resourceLoader.getResource("classpath:samples/ERC000024/SAMEA110262237-bsd.json").getFile();
+      JsonNode sampleJson = objectMapper.readValue(file, JsonNode.class);
+
+      requestNode = objectMapper.createObjectNode();
+      requestNode.set("data", sampleJson);
+      requestNode.set("schema", schemaJson);
+
+      response = webClient.post()
+          .uri("http://localhost:3020/validate")
+          .bodyValue(requestNode)
+          .retrieve()
+          .toEntity(JsonNode.class)
+          .toFuture()
+          .get();
```

Comment:
This bit is very similar to the previous test code. I would recommend to extract code to functions. This makes tests more readable, because they are shorter in  are written in a higher level of abstraction, therefore do not overflow the reader with implementation details. Also, when you have functions, you can reuse them in other tests, so it becomes easy to add more tests.